src/rp2_common/pico_crt0/crt0.S

```diff
@@ -11,6 +11,7 @@
 
 #include "hardware/regs/addressmap.h"
 #include "hardware/regs/sio.h"
+#include "hardware/irq.h"
```

Comment:
Looks like a new dependency on the `hardware_irq` library, which is not reflected in the CMake.

Ideally I'd like to be able to pull in the crt0 without the rest of the SDK. `hardware_irq` pulls in `pico_sync` etc

Reply:
yeah, actually it was a bit of a circular nightmare because now the defines in hardware_irq.h are dependent on the crt0 vars... I'll actually go to what i was half way thru doing (which is to make hardware_irq dependent on the crt0.S headers - which I had avoided since it is hardware->pico - but i hadn't realized that already dependended on pico headers.

Reply:
i have created a new pico_platfrom_common since we didn't have a good place to put common code between different pico_platforms (and this header is a good place to put these common defines)

Reply:
^ note that this has deduplicated some code that was in both rp2350/pico_platform and rp2040/pico_platform